### ISSUE TITLE
Clear designation of train/validation/test sets

### ISSUE THREAD

**Issue by kdesoto-astro** [kdesoto-astro/survey-agnostic-sn-vae#3]
From reviewer:

_"Given the 9:1 train/validation split there is no test set, and you used the validation set for VAE training. While this may be reasonable for the VAE component alone, this makes me concerned for the classification experiment. Was this on the train or validation set? Either way, these results do not reflect performance in the wild. Your model has already “seen” every SN in the dataset. This should be performed on a true hold out test set, since the end goal is to classify new observations."__

IIRC the same validation set from VAE training was used as the test set for classification. Because hyperparameter optimization is done wtih the validation set, we do need a distinct "test" set that propagates all the way through to classification.
